src/mockttp.ts

```diff
@@ -415,6 +416,23 @@ export interface Mockttp {
      */
     on(event: 'client-error', callback: (error: ClientError) => void): Promise<void>;
 
+    /**
+     * Subscribe to hear about requests that fail before successfully sending their
+     * initial parameters (the request line & headers). This will fire for requests
+     * that drop connections early, send invalid or too-long headers, or aren't
+     * correctly parseable in some form.
+     *
+     * This is only useful in some niche use cases, such as logging all requests
+     * seen by the server, independently of the rules defined.
+     *
+     * The callback will be called asynchronously from request handling. This function
+     * returns a promise, and the callback is not guaranteed to be registered until
+     * the promise is resolved.
+     *
+     * @category Events
+     */
+    on(event: 'handle-error', callback: (error: ErrorLike) => void): Promise<void>;
```

Comment:
Do you think it would be useful to include the request that failed here too as a second parameter, something like `(error, req) => { ... }`?

That would be as an `InitiatedRequest` I think, since we don't actually know if it completed. Or maybe we can do `InitiatedRequest | CompletedRequest`, I'm just not sure if we have an easy way to provide the completed data only if it's ready, without waiting if it's not...